src/main/java/org/dbos/apiary/mysql/MysqlConnection.java

```diff
@@ -27,21 +31,60 @@ public class MysqlConnection implements ApiarySecondaryConnection {
     public Percentile upserts = new Percentile();
     public Percentile queries = new Percentile();
     public Percentile commits = new Percentile();
-
+    public AtomicInteger groupFlushLogs = new AtomicInteger(0);
     private final MysqlDataSource ds;
     private final ThreadLocal<Connection> connection;
     private final ThreadLocal<Connection> commitConnection;
     private boolean delayLogFlush = false;
     private final Map<String, Map<String, Set<Long>>> committedWrites = new ConcurrentHashMap<>();
     private final Lock validationLock = new ReentrantLock();
-
+    private static ConcurrentLinkedQueue<CountDownLatch> groupFlushRequests = new ConcurrentLinkedQueue();
+    private static AtomicBoolean groupFlusherStarted = new AtomicBoolean(false);
+    public void enableSlowQueryLog() {
+        try {
+            Connection conn = ds.getConnection();
+            Statement s = conn.createStatement();
+            s.execute("SET GLOBAL slow_query_log=1");
+        } catch (SQLException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+    }
     public void enableDelayedFlush() {
         this.delayLogFlush = true;
         try {
+            
             Connection conn = ds.getConnection();
             conn.setAutoCommit(true);
             Statement s = conn.createStatement();
             s.execute("SET GLOBAL innodb_flush_log_at_trx_commit=0;");
+            if (groupFlusherStarted.compareAndSet(false, true)) {
+                new Thread( () -> {
+                    logger.info("GroupDelayFlush Thread started");
+                    while (true) {
+                        List<CountDownLatch> latches = null;
+                        while (groupFlushRequests.isEmpty() == false) {
```

Comment:
This code can be cleaned up, for example checking (!groupFlushRequests.isEmpty()).  I also don't understand how this optimization works--is it very likely that two concurrent transactions will end up in the same batch, given that you start processing a batch almost immediately after adding its first item?